Miscellaneous editorial fixes (#52).

skynavga opened the pull request:
Closes #52.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/tt-profile-registry/pull/55.html" title="Last updated on Feb 28, 2019, 3:02 PM UTC (443cd2a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/tt-profile-registry/55/acd0e10...443cd2a.html" title="Last updated on Feb 28, 2019, 3:02 PM UTC (443cd2a)">Diff</a>

skynavga committed:
Miscellaneous editorial fixes (#52).

nigelmegitt commented on index.html:
```diff
@@ -60,6 +63,80 @@
           }],
 
           localBiblio: {
+            "IMSC1.0": {
```
This is in Specref as [`[ttml-imsc1.0.1-20180424]`](https://www.specref.org/?q=ttml-imsc1.0.1-20180424) which can be used here instead of defining it locally. For conciseness it can also be aliased to `IMSC1.0` for local use, for example:

```es
"IMSC1.0": {
aliasOf: "ttml-imsc1.0.1-20180424"
},
```

However I would prefer to use `IMSC1.0.1` rather than `IMSC1.0` - I'd take views on it though because I can see arguments both ways - any thoughts @palemieux ?

palemieux replied:
> I would prefer to use IMSC1.0.1

+1

skynavga replied:
ok, will change; (I mean I will change to use IMSC1.0.1, not use an alias to a specref entry - see below);

nigelmegitt replied:
Fixed in 5483f8e

nigelmegitt commented on index.html:
```diff
@@ -60,6 +63,80 @@
+              status: "REC",
+              date: "24 April 2018"
+            },
+            "IMSC1.1": {
```
This is in Specref as [`[ttml-imsc1.1-20181108]`](https://www.specref.org/?q=ttml-imsc1.1-20181108) which should be aliased instead of a local definition.

skynavga replied:
I am completely opposed to making any use of specref references for the simple reasons that (1) they can (and do) change from under us without notice, thus making any document that employs them unstable, and (2) they do not contain the same information that we employ in our primary TTML specifications. We can discuss this further in teleconference.

nigelmegitt replied:
Using Specref is normal practice in W3 specs. If you want to change the entries, or have visibility of proposed change, this is done at https://github.com/tobie/specref .

skynavga replied:
That does not fix the document stability issue. This is really a pure editorial issue in the domain of the editors. Our choice is to use inline definitions to avoid instability due to unexpected and unannounced changes in specref over which  we have no editorial control.

nigelmegitt replied:
That's not correct: the document is stable with respect to W3C specifications because the W3C /TR space spec info provided by specref is sourced from https://www.w3.org/2002/01/tr-automation/tr.rdf . If you think the Specref information isn't correct then we need to raise an issue with the team who maintain tr.rdf (or the tool that updates it on spec publication).

skynavga replied:
I disagree. The sprecref data is not under direct control of the editors of this document nor is under the process control of this group; therefore, it is unstable and unfixed. If you like, we can simply convert this document to HTML5 or XML Doc format and stop using respec, but I will not agree to use respec if it forces us to use specref data to resolve bibliographic data. I note once again that this decision is in the domain of the editors (absent formal TTWG policy to the contrary).

nigelmegitt replied:
In my view consistency across W3C documents takes precedence over consistency across TTWG documents; since the data is provided by W3C, we should use it.  Furthermore, I've just made the change locally to switch all the W3C spec references into aliases and the only differences I can see are:

1. the dates do not consistently zero-pad the day part to a two digit number in the text description of the document being referenced.
2. for SDP-US the phrase "W3C Note" becomes "W3C Working Group note"

So there does not seem to be a substantive problem with changing this, and we could consider raising a query about the date format.

Since I've made the change, I'd be happy to push it directly if you'd withdraw your objection @skynavga ?

skynavga replied:
No, I do not withdraw my objection. As you have shown, even your attempt to synchronize the content was unsuccessful. The larger issue for me is that you are personally attempting to impose editorial control on a document for which you are not an editor, and you are blocking progress on the part of the editor's work.

nigelmegitt replied:
> As you have shown, even your attempt to synchronize the content was unsuccessful.

I drew the opposite conclusion from the minor nature of the deltas! Essentially, I thought my attempt was successful and that the differences were so small as to be not worth holding out for.

Regarding "editorial control" I disagree, but as you've noted before, Github spec repository pull requests and issues aren't a good place to discuss process.

skynavga replied:
I have looked at specref.org. It is not affiliated with the W3C and appears to impose copyright status on the content that may be used for specification data. As such, and here I am speaking with both my editor and AC representative hats, I cannot accept content that is arbitrarily inserted into our working group documents for which we as a TTWG and the W3C itself has no control and may be in violation of IPR policies.

nigelmegitt replied:
@plehegar can you please provide guidance about https://github.com/w3c/tt-profile-registry/pull/55#discussion_r251211105 ? Specref seems to be used endemically in W3C specs and that seems like a good thing to me, but @skynavga is raising an issue that it is not suitable for use in this way.

A couple of other points worth considering @skynavga :
1. When the spec gets published to /TR the references etc are frozen, i.e. it's the respec formatted output that is uploaded, not the version with the respec script running. This means that even if the reference changes in Specref it won't magically break/fix documents in /TR.
2. Specref cannot claim copyright on the data it is providing, which is clearly owned by whatever organisations are publishing the specifications themselves. The only way it might have copyright is if the information is deliberately incorrect, which I believe is never the case.

nigelmegitt commented on index.html:
```diff
@@ -60,6 +63,80 @@
+              status: "REC",
+              date: "08 November 2018"
+            },
+            "TTML1-1e": {
```
This is in Specref as [`[ttaf1-dfxp-20101118]`](https://www.specref.org/?q=ttaf1-dfxp-20101118) which should be aliased instead of a local definition.

skynavga replied:
ditto

nigelmegitt commented on index.html:
```diff
@@ -60,6 +63,80 @@
+              status: "REC",
+              date: "18 November 2010"
+            },
+            "TTML1-2e": {
```
This is in Specref as [`[ttml1-20130924]`](https://www.specref.org/?q=ttml1-20130924) which should be aliased instead of a local definition.

skynavga replied:
ditto

nigelmegitt commented on index.html:
```diff
@@ -60,6 +63,80 @@
+              status: "REC",
+              date: "24 September 2013"
+            },
+            "TTML1-3e": {
```
This is in Specref as [`[ttml1-20181108]`](https://www.specref.org/?q=ttml1-20181108) which should be aliased instead of a local definition.

skynavga replied:
ditto

nigelmegitt commented on index.html:
```diff
@@ -60,6 +63,80 @@
+              status: "REC",
+              date: "08 November 2018"
+            },
+            "TTML1-SDP-US": {
```
This is in Specref as [`[ttml10-sdp-us-20130205]`](https://www.specref.org/?q=ttml10-sdp-us-20130205) which should be aliased instead of a local definition.

skynavga replied:
ditto

nigelmegitt commented on index.html:
```diff
@@ -60,6 +63,80 @@
+              status: "WG-NOTE",
+              date: "05 February 2013"
+            },
+            "TTML2-1e": {
```
This is in Specref as [`[ttml2-20181108]`](https://www.specref.org/?q=ttml2-20181108) which should be aliased instead of a local definition.

skynavga replied:
ditto

nigelmegitt commented on index.html:
```diff
@@ -73,698 +150,734 @@
+      document, in
+      a <a href="https://www.w3.org/TR/ttml1/#vocabulary-profiles">TTML
+      Profile Definition Document</a>, inline within a TTML Document Instance,
+      or by using a combination of these methods. This document document only
```
s/document document/document

nigelmegitt replied:
Fixed in 5483f8e

nigelmegitt commented on index.html:
```diff
@@ -73,698 +150,734 @@
+      <p>A MIME type and subtype
+        combination, <code>application/ttml+xml</code>, was initially
+        defined in [[TTML1-1e]], Appendix C, and subsequently updated
+        by [[TTML1-2e]] and [[TTML1-3e]] in order to identify the
```
We have not updated the media type registration based on TTML1-3e because _this_ document became the registration document in place of TTML1-2e.

nigelmegitt replied:
This comment not resolved yet.

nigelmegitt commented on index.html:
```diff
@@ -73,698 +150,734 @@
+        other W3C and non-W3C specifications define a number of
+        processor profiles which specify requirements on processors
+        that may decode and process a TTML document. This document
+        augments the definition of this MIME type by specifying a
```
This isn't right - the IANA [registration](https://www.iana.org/assignments/media-types/application/ttml+xml) points at this document, so the published version of this document (i.e. https://www.w3.org/TR/ttml-profile-registry/) _is_ the definition of the MIME type and therefore cannot augment it.

There is some ambiguity introduced by the lines below ([line 241](https://github.com/w3c/tt-profile-registry/pull/55/files#diff-eacf331f0ffc35d4b482f1d15a887d3bR241) in this pull request right now) that say "This section updates the media type", which is a relative statement compared to the previous registration, but I suggest we leave that as is.

skynavga replied:
we need to discuss this in teleconference; if we can't update the definition, then the `codecs` parameter will not have normative status

nigelmegitt replied:
sure, I'll add it to the agenda at w3c/ttwg#11 

Why do we need to update the definition?

skynavga replied:
Addressed in https://github.com/w3c/tt-profile-registry/pull/55/commits/abc547e4d41e27c4ce0de4477443efba6b826f26 per a previous telecon.

nigelmegitt replied:
This doesn't look addressed to me. Suggest rewording to:

> After the original MIME type was registered, it was augmented to include a `codecs` parameter. This document includes the public register of compact (short) identifiers to be used with this parameter during the exchange of TTML document entities.

nigelmegitt reviewed (changes requested):
Mostly editorial comments, such as aliasing specs defined in Specref rather than redefining them, and a typo, but the language about the Purpose of the document needs editing - it isn't right as is.

skynavga committed:
Address review comments (https://github.com/w3c/tt-profile-registry/pull/55#pullrequestreview-189824826) (#55,#52).

nigelmegitt committed:
Switch to specref aliases for W3C documents

Avoids duplication of details and uses the canonical reference for /TR documents.

skynavga committed:
Address https://github.com/w3c/tt-profile-registry/pull/55#discussion_r245676736.

skynavga committed:
Merge remote-tracking branch 'origin/gh-pages' into issue-52-misc-editorial

skynavga committed:
Merge pull request #59 from w3c/issue-52-switch-to-specref-aliases

Switch to specref aliases for W3C documents

skynavga commented:
@nigelmegitt merged from PR #59; please approve for merger to gh-pages

nigelmegitt commented on index.html:
```diff
@@ -73,698 +97,730 @@
-      <p>This document defines the <a href='#mediatype'><code>application/ttml+xml</code></a> media type and provides a registry of identified TTML processor profiles. A processor profile is an identified reference to a set of capabilities that a processor supports, which may be defined in a specification document, a <a href='https://www.w3.org/TR/ttml1/#vocabulary-profiles'>TTML Profile Definiti ...
+      <h2>Abstract</h2>
+      <p>This document defines
+      the <a href='#mediatype'><code>application/xml+ttml</code></a> media
```
This edit reverts a previous fix: it should be `ttml+xml`

nigelmegitt commented on index.html:
```diff
@@ -73,698 +97,730 @@
+        <p>In the context of this registry, the term <i>profile</i>
+          means <i>processor profile</i>. It explicitly does not
+          mean <i>content profile</i>. That is, nothing about the use
+          of the <code>profile</code> parameter described here is
```
needs to mention `codecs` too, e.g. replace with:

```
of the <code>profile</code> or <code>codecs</code> parameters described here is
```

skynavga committed:
Attempt to address outstanding review comments.

skynavga commented:
@nigelmegitt I have attempted to address your comments at https://github.com/w3c/tt-profile-registry/pull/55#pullrequestreview-206754013 (and make a few respec warning fixes); please take a look

nigelmegitt commented on index.html:
```diff
@@ -73,698 +97,710 @@
+        and, in conformance with
+        <a href="https://www.ietf.org/rfc/rfc4288.txt">BCP 13</a>
+        and <a href="http://www.w3.org/2002/06/registering-mediatype.html">W3CRegMedia</a>,
+        serves as the registration thereof with the Internet Assianged
```
typo: s/Assianged/Assigned

nigelmegitt reviewed (changes requested):
Sorry, spotted a typo, will fix in place and then approve.

nigelmegitt committed:
Fix typo

nigelmegitt reviewed (approved):
Thanks @skynavga, looking good to me now.
